spec/requests/portfolio_items_spec.rb

```diff
@@ -232,4 +232,99 @@
       end
     end
   end
+
+  describe "copying portfolio items" do
+    let(:copy_portfolio_item) do
+      post "#{api("1.0")}/portfolio_items/#{portfolio_item.id}/copy", :params => params, :headers => default_headers
+    end
+
+    context "when copying into the same portfolio" do
+      let(:params) { { :portfolio_id => portfolio.id } }
+
+      before do
+        copy_portfolio_item
+      end
+
+      it "returns a 200" do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "returns a copy of the portfolio_item" do
+        expect(json["description"]).to eq portfolio_item.description
+        expect(json["owner"]).to eq portfolio_item.owner
+        expect(json["workflow_ref"]).to eq portfolio_item.workflow_ref
+      end
+
+      it "modifies the name to not collide" do
+        expect(json["name"]).to_not eq portfolio_item.name
+        expect(json["name"]).to match(/^Copy of.*/)
+      end
+    end
+
+    context "when copying with a specified name" do
+      let(:params) { { :portfolio_item_name => "NewPortfolioItem" } }
+
+      it "returns the name specified" do
+        copy_portfolio_item
+        expect(json["name"]).to eq params[:portfolio_item_name]
+      end
+    end
+
+    context "when copying an item multiple times" do
+      let(:params) { { :portfolio_id => portfolio.id } }
+      let!(:another_portfolio_item) do
+        create(:portfolio_item,
```

Comment:
I wonder if we could move some of these specific naming specs into a spec specific to the new copy name service?  I see value in validating the copy endpoint in this spec and testing the functionality in the service spec.

Reply:
Good idea, I moved the specific naming ones to the service spec. I left the ones in that tested the endpoint and error handling in.